src/library_syscall.js

```diff
@@ -307,7 +307,9 @@ var SyscallsLibrary = {
   },
   __syscall_unlink: function(path) {
     path = SYSCALLS.getStr(path);
-    FS.unlink(path);
+    if (path.length) {
+      FS.unlink(path);
```

Comment:
According the docs for unlink the (at least the C library version) the it is supposed to return `ENOENT` when the file does not exist.

So perhaps this would be a better solution:

```
if (!path.length) -{{{ cDefine('ENOENT') }}};
```

Alternatively perhaps modify the `FS.unlink` code to handle this case?

Reply:
Yes I think maybe the better fix is to add:

```
      if (!parent) {                                                               
        return -{{{ cDefine('ENOENT') }}};                                                                     
      } 
```

To the `unlink` function in `library_fs.js`